framework/src/engine/generator/network_endpoint.ts

```diff
@@ -226,19 +226,44 @@ export class NetworkEndpoint extends BaseNetworkEndpoint {
 			);
 
 			try {
-				for (const transactionBytes of transactionsData.transactions) {
-					const transaction = Transaction.fromBytes(transactionBytes);
-					await this._receiveTransaction(transaction);
-				}
+				await Promise.all(
+					transactionsData.transactions.map(async transactionBytes => {
```

Comment:
why was it changed to `Promise.all`?

Reply:
To run the validation of each transaction concurrently. But I can revert 🙂

Reply:
`fromBytes` and `validate` are synchronous function, so probably it doesn't make so much sense. (and if one fails, it's ok to invalidate all the remaining message)